### ISSUE TITLE
Build fails with map(provider(java.util.Set)) not supported after upgrading to AGP 7.4.0-rc01 and 6.1.0

### ISSUE THREAD

**Issue by leinardi** [usefulness/easylauncher-gradle-plugin#387]
After upgrading the Android Gradle Plugin from `7.4.0-beta05` to `7.4.0-rc01` the build fails with the error:

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:mapDebugSourceSetPaths'.
> Querying the mapped value of map(provider(java.util.Set)) before task ':app:easylauncherDebug' has completed is not supported
```

Easy launcher version 6.1.0 (version 6.0.0 seems to not be affected).
Gradle 7.5.1 (tested also with 7.6.)
Gradle command:  `./gradlew assembleDebug assembleDebugAndroidTest`

**Comment by mateuszkwiecinski** [comment]
Hey 👋 Thanks for reporting the issue. I'm unable to confirm that on my end right now, but it seems like that's a known bug of AGP which became extremely unstable recently 🤷‍♂️ https://issuetracker.google.com/issues/261329823

As far as I saw, there will be another RC of 7.4.0, so hopefully it will be fixed on their side.

**Comment by casperbang** [comment]
I can confirm hitting this issue as well. Better to stay on 7.4.0-beta05 and hope 7.4.0-rc02 remedies the situation.

**Comment by luanbarbosa-sanoma** [comment]
Same for me. With AGP `8.0.0-alpha09` version `6.1.0` fails but `6.0.0` works fine.

**Comment by mateuszkwiecinski** [comment]
I'll share a small update:
According to the [issue](https://issuetracker.google.com/issues/261329823) linked above, the fix will be available in the following versions: 
<img width="288" alt="image" src="https://user-images.githubusercontent.com/36954793/209081518-f9f021e3-4304-4fd9-a54e-3beb778f0116.png">
I did a [smoke test](https://github.com/usefulness/easylauncher-gradle-plugin/pull/392) and latest `8.0.0-alpha10` seems to be working fine again, 7.4.0-rc2 is still affected. As far as I understand, the bug will land in a "stable" release, and we'll have to wait for a patch release 🤷

**Comment by ivannarino** [comment]
I can confirm the bug still happens in 7.4.0 release with 6.1.0

**Comment by mateuszkwiecinski** [comment]
It seems like the issue has been fixed in AGP `7.4.1`, or at least I'm not able to reproduce it anymore 👀 

I'm going to close this issue, but I'll leave a reminder that:`7.4.1` still has this bug: https://issuetracker.google.com/issues/244063664 and requires a [workaround](https://issuetracker.google.com/issues/244063664#comment5) for it, but other than that it seems to work just fine 👀

**Comment by leinardi** [comment]
I can confirm that version 6.1.0 and AGP 7.4.1 works fine now, but `android.disableResourceValidation=true` is still required.
